sysid-application/src/main/native/cpp/analysis/FilteringUtils.cpp

```diff
@@ -96,10 +95,14 @@ units::second_t sysid::TrimStepVoltageData(std::vector<PreparedData>* data,
           return std::abs(entry.acceleration) > accelNoiseFloor;
         });
 
-    // Calculate default duration
-    settings->stepTestDuration =
-        std::min(endIt->timestamp - data->front().timestamp + minStepTime + 1_s,
-                 maxStepTime);
+    if (endIt != data->rend()) {
```

Comment:
Why `rend()` instead of `end()`?

Reply:
I think endIt is a reverse iterator. Does it not make a difference if I use `end()`?

Reply:
Oh, I see. In that case, we should probably change this:
```cpp
    auto endIt = std::find_if(
        std::reverse_iterator{data->end()},
        std::reverse_iterator{data->begin()}, [&](const PreparedData& entry) {
          return std::abs(entry.acceleration) > accelNoiseFloor;
        });
```
to this:
```cpp
    auto endIt = std::find_if(
        data->rend(), data->rbegin(), [&](const PreparedData& entry) {
          return std::abs(entry.acceleration) > accelNoiseFloor;
        });
```

Reply:
Wouldn’t that be rbegin(), rend()?  The rbegin() function returns the reverse iterator starting at the end of the container.

Reply:
Yeah, it is that. It doesn't compile otherwise.